config/samples/oauth-token-exchange/tools-call-auth.yaml

```diff
@@ -0,0 +1,67 @@
+apiVersion: kuadrant.io/v1
+kind: AuthPolicy
+metadata:
+  name: mcps-auth-policy
+  namespace: gateway-system
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: Gateway
+    name: mcp-gateway
+    sectionName: mcps
+  rules:
+    authentication: #validates the token
+      'keycloak':
+        jwt:
+          issuerUrl: http://keycloak.keycloak.svc.cluster.local/realms/mcp
+    metadata:
+      token-exchange:
+        http:
+          url: http://keycloak.keycloak.svc.cluster.local/realms/mcp/protocol/openid-connect/token # FIXME: this will make the token exchange request fail with 'Invalid Token' due to mismatching Keycloak hostname compared to what's in the `iss` claim of the subject token
+          method: POST
+          headers:
+            Authorization:
+              value: 'Basic YXV0aG9yaW5vOnNlY3JldA==' # authorino:secret
+          bodyParameters:
+            grant_type:
+              value: urn:ietf:params:oauth:grant-type:token-exchange
+            subject_token:
+              expression: request.headers['authorization'].split('Bearer ')[1] # remove "Bearer "
```

Comment:
This line upsets me because I don't know what this language will do with `[1]` if the split generates 0 or 1 items.  The authorization header comes from outside the system and therefore can be controlled by an attacker.  Perhaps this is no problem, but I couldn't figure out what language is used here (Rego expression?) and how to test various inputs in a playground to ensure safety.

Reply:
This is [Common Expression Language (CEL)](https://cel.dev).

In general, an expression like this in CEL could be a problem if the split returned less than 2 items. However, in this particular instance, it is not a problem and it will never be a problem.

The reason is because the CEL expression is evaluated in the metadata phase of the AuthPolicy, which means it passed the authentication phase. The authentication phase won't ever succeed if the value of the `Authorization` header is not exactly equal to `Bearer <valid jwt>`.